.github/workflows/cd.yaml

```diff
@@ -82,19 +106,80 @@ jobs:
 
-            - name: Run Smoke Test
+            - name: Run Smoke Test (NPM ES)
+              env:
+                  URL: ${{ secrets.SMOKE_URL }}
+                  MONITOR: ${{ secrets.SMOKE_MONITOR }}
+                  ENDPOINT: ${{ secrets.SMOKE_ENDPOINT }}
+                  NAME: ${{ secrets.SMOKE_MONITOR_NAME }}
+                  INSTALL_METHOD: 'NPM-ES'
+              run: npm run smoke:headless
+              timeout-minutes: 10
+
+            - name: Fetch AWS Credentials for Smoke Test
+              run: |
+                  export AWS_ROLE_ARN=${{ secrets.SMOKE_TEST_ROLE }}
+                  export AWS_WEB_IDENTITY_TOKEN_FILE=/tmp/awscreds
+                  export AWS_DEFAULT_REGION=us-east-1
+
+                  echo AWS_WEB_IDENTITY_TOKEN_FILE=$AWS_WEB_IDENTITY_TOKEN_FILE >> $GITHUB_ENV
+                  echo AWS_ROLE_ARN=$AWS_ROLE_ARN >> $GITHUB_ENV
+                  echo AWS_DEFAULT_REGION=$AWS_DEFAULT_REGION >> $GITHUB_ENV
+
+                  curl -H "Authorization: bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" "$ACTIONS_ID_TOKEN_REQUEST_URL" | jq -r '.value' > $AWS_WEB_IDENTITY_TOKEN_FILE
+              if: always()
+
+            - name: Run Smoke Test (NPM CJS)
+              env:
+                  URL: ${{ secrets.SMOKE_URL }}
+                  MONITOR: ${{ secrets.SMOKE_MONITOR }}
+                  ENDPOINT: ${{ secrets.SMOKE_ENDPOINT }}
+                  NAME: ${{ secrets.SMOKE_MONITOR_NAME }}
+                  INSTALL_METHOD: 'NPM-CJS'
+              run: npm run smoke:headless
+              timeout-minutes: 10
+              if: always()
+
+            - name: Fetch AWS Credentials for Smoke Test
+              run: |
+                  export AWS_ROLE_ARN=${{ secrets.SMOKE_TEST_ROLE }}
+                  export AWS_WEB_IDENTITY_TOKEN_FILE=/tmp/awscreds
+                  export AWS_DEFAULT_REGION=us-east-1
+
+                  echo AWS_WEB_IDENTITY_TOKEN_FILE=$AWS_WEB_IDENTITY_TOKEN_FILE >> $GITHUB_ENV
+                  echo AWS_ROLE_ARN=$AWS_ROLE_ARN >> $GITHUB_ENV
+                  echo AWS_DEFAULT_REGION=$AWS_DEFAULT_REGION >> $GITHUB_ENV
+
+                  curl -H "Authorization: bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" "$ACTIONS_ID_TOKEN_REQUEST_URL" | jq -r '.value' > $AWS_WEB_IDENTITY_TOKEN_FILE
+              if: always()
+
+            - name: Run Smoke Test (CDN)
               env:
                   URL: ${{ secrets.SMOKE_URL }}
                   MONITOR: ${{ secrets.SMOKE_MONITOR }}
                   ENDPOINT: ${{ secrets.SMOKE_ENDPOINT }}
                   NAME: ${{ secrets.SMOKE_MONITOR_NAME }}
+                  INSTALL_METHOD: 'CDN'
               run: npm run smoke:headless
+              if: always()
 
             - name: Publish to NPM
               run: npm publish
               env:
                   NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
 
+            - name: Build Smoke Test Application - NPM/ES (Post NPM Release)
```

Comment:
question: I understand the idea behind this, but is it necessary? Technically if the packed ES/CJS tests succeed, wouldn't NPM most likely succeed as well?

Reply:
question: Based on the sample workflow you've provided, it seems we continued to run all three different types of smoke tests even if the ES one failed - did the workflow stop at NPM because it is a gamma stage and the credentials were wrong, or is it because the tests have failed?

Reply:
> question: I understand the idea behind this, but is it necessary? Technically if the packed ES/CJS tests succeed, wouldn't NPM most likely succeed as well?

You are correct that theoretically its unlikely there would be an issue. 

This is more of a sanity check that the publishing process is done correctly and replicates the behavior seen when installing the package locally during development. It decreases the overall time to mitigation if the release had an issue or for some reason NPM install is not working correctly